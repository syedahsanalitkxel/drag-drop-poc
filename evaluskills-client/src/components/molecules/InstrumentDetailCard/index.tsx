import React, { ReactNode } from 'react';

interface Props {
  header: string;
  children: {
    content: ReactNode;
    actions?: ReactNode;
    bar?: ReactNode;
  };
}

const InstrumentCard: React.FunctionComponent<Props> = ({ header, children }) => {
  const { content, actions, bar } = children;

  return (
    <div className="es-card">
      <div className="row">
        <div className="col-md-8">
          <p className="es-card-title">
            {header}&nbsp;<span className="badge badge-warning">Draft</span>
          </p>
          <div className="row es-card-info">{content}</div>
        </div>
        <div className="col-md-2 d-flex align-items-center">
          <div className="es-card-button">{actions}</div>
        </div>
        <div className="col-md-2 d-flex align-items-center">
          <div className="assesment-item-button">{bar}</div>
        </div>
      </div>
    </div>
  );
};
export default InstrumentCard;
