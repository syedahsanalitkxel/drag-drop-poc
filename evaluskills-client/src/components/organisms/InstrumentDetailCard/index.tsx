import React, { Fragment, useState } from 'react';
import InstrumentCard from '../../molecules/InstrumentCard';
import ParticipantInterface from '../../../interfaces/Participant';
import AddEvaluator from '../../pages/AddEvaluator';

interface ListCardProps {
  participants: ParticipantInterface[];
  titleKey: string;
  view?: (evaluationId: string) => void;
  actionHandler?: (values?: any, evaluationId?: string, action?: string, token?: string) => void;
}

const InstrumentListCard: React.FunctionComponent<ListCardProps> = ({
  titleKey,
  participants,
  view,
  actionHandler,
}) => {
  const [evaluatorModalVisible, setEvaluatorModalVisible] = useState(false);
  const [evaluator, setEvaluator] = useState({});
  const [token, setToken] = useState('');

  const addEvaluator = (id: string, action: string, evalToken?: string) => {
    if (action === 'add' && actionHandler && evalToken) {
      setEvaluatorModalVisible(true);
      setToken(evalToken);
    } else if (action === 'remove' && actionHandler) {
      actionHandler({}, id, action, evalToken);
    }
  };

  const toggleAddEvaluatorModal = () => {
    setEvaluatorModalVisible(!evaluatorModalVisible);
  };

  function submitHandler(values: any) {
    if (actionHandler) {
      setEvaluatorModalVisible(false);
      actionHandler(values, '', 'add', token);
    }
  }

  function cancelHandler() {
    setEvaluatorModalVisible(false);
  }

  return (
    <React.Fragment>
      <div className="ibox ">
        <div className="ibox-content p-0">
          <table className="table panel-group" id="accordions">
            <thead>
              <tr>
                <th>Participants</th>
                <th>Email</th>
                <th>No. of Evaluators</th>
                <th>Progress</th>
                <th>Status</th>
                <th>Individual Results</th>
                <th />
              </tr>
            </thead>
            <tbody>
              {participants &&
                participants.map((list, index) => (
                  <InstrumentCard key={index} item={list} view={view} addEvaluator={addEvaluator} />
                ))}
            </tbody>
          </table>
        </div>
      </div>
      <React.Fragment>
        <AddEvaluator
          visible={evaluatorModalVisible}
          toggle={toggleAddEvaluatorModal}
          name="Add"
          FormValues={evaluator}
          submitHandler={submitHandler}
          cancelHandler={cancelHandler}
        />
      </React.Fragment>
    </React.Fragment>
  );
};

export default InstrumentListCard;
