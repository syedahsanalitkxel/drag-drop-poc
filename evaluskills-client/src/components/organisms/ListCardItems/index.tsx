import React from 'react';

import { IconProp } from '@fortawesome/fontawesome-svg-core';
import styled from 'styled-components';
import PageBody from '../../atoms/PageBody';
import { BadgeTypes } from '../../../enums';
import Checkbox from '../../atoms/CheckBox';
import IconButton from '../../atoms/IconButton';
import EmptyPage from '../../atoms/EmptyPage';
import LabelGroup from '../../atoms/LabelGroup';
import ItemCard from '../../molecules/ItemCard';
import { TemplateItem } from '../../../modules/InstrumentTemplate/interface';

interface ListCardProps {
  listData: any[];
  titleKey: string;
  edit?: (id: string) => void;
  remove?: (id: string) => void;
  checkbox?: boolean;
  handleCheckbox?: (templateItem: TemplateItem) => void;
  checkedItems?: string[];
}

const CheckboxContainer = styled.div`
  position: absolute;
  right: 0;
  top: 30%;
`;

const ListCardItems: React.FunctionComponent<ListCardProps> = ({
  titleKey,
  listData,
  edit,
  remove,
  checkbox,
  checkedItems,
  handleCheckbox,
}) => {
  const actionHandler = (assessmentId: string) => (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    if (event.currentTarget.name === 'edit') {
      if (edit) {
        edit(assessmentId);
      }
    } else if (event.currentTarget.name === 'delete') {
      if (remove) {
        remove(assessmentId);
      }
    }
  };

  const renderContent = (category: string, type: string, competency: string) => (
    <React.Fragment>
      <div className="col-md-3 p-l-0">
        <LabelGroup label="Category" value={category} />
      </div>
      <div className="col-md-3 border-left text-center">
        <LabelGroup label="Type" value={type} badge={BadgeTypes.PRIMARY} />
      </div>
      <div className="col-md-4 border-left text-center">
        <LabelGroup label="Competency" value={competency} />
      </div>
    </React.Fragment>
  );

  const renderActions = (id: string, item: any) => {
    const renderActionButton = (name: string, text: string, icon: IconProp, className: string) => (
      <IconButton name={name} icon={icon} className={className} actionHandler={actionHandler(id)}>
        {text}
      </IconButton>
    );

    if (checkbox) {
      return (
        <CheckboxContainer className="pull-right">
          <Checkbox
            name="templateItem"
            value={id}
            isChecked={checkedItems && !!checkedItems.find(checkedItem => checkedItem === id)}
            onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
              e.preventDefault();
              if (handleCheckbox) {
                handleCheckbox(item);
              }
            }}
          >
            {' '}
          </Checkbox>
        </CheckboxContainer>
      );
    }

    return (
      <React.Fragment>
        {edit && renderActionButton('edit', 'Edit', 'edit', 'btn-outline btn-primary')}
        {remove && renderActionButton('delete', 'Delete', 'trash', 'btn-default')}
      </React.Fragment>
    );
  };

  function renderAllCards(item: any) {
    const content =
      (item.category || item.type || item.competency) && renderContent(item.category, item.type, item.competency);

    const actions = renderActions(item.id || item.itemId, item);

    return (
      <ItemCard key={item.id || item.itemId} header={item[titleKey]}>
        {{ content, actions }}
      </ItemCard>
    );
  }

  return (
    <React.Fragment>
      {listData.length > 0 ? (
        listData && listData.map(renderAllCards)
      ) : (
        <PageBody card={true} wrapper={true} className="m-t-15">
          <EmptyPage />
        </PageBody>
      )}
    </React.Fragment>
  );
};

export default ListCardItems;
