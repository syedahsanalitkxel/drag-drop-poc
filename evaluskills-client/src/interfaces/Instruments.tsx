export interface ClientInstruments {
  title: string;
  noOfParticipants: string;
  noOfEvaluations: string;
  noOfAssessmentItems: string;
  id: string;
}

export default interface Insturments {
  instructionTitle: string;
  editorState?: any;
  componentName: string;
}
