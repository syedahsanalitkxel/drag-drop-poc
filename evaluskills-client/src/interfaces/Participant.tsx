export default interface ParticipantInterface {
  name: string;
  email: string;
  noOfEvaluators: string;
  id: string;
  progress: string;
  status: string;
}
