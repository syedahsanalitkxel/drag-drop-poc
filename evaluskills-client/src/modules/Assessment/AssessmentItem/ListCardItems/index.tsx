import React from 'react';

import { IconProp } from '@fortawesome/fontawesome-svg-core';
import styled from 'styled-components';
import PageBody from '../../../../components/atoms/PageBody';
import { BadgeTypes } from '../../../../enums';
import Checkbox from '../../../../components/atoms/CheckBox';
import IconButton from '../../../../components/atoms/IconButton';
import EmptyPage from '../../../../components/atoms/EmptyPage';
import LabelGroup from '../../../../components/atoms/LabelGroup';
import ItemCard from '../../../../components/molecules/ItemCard';
import { TemplateItem } from '../../../InstrumentTemplate/interface';
import { getActiveClient, USER_ROLE } from '../../../../utils';
import classNames from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface ListCardProps {
  listData: any[];
  titleKey: string;
  edit?: (id: string) => void;
  remove?: (id: string) => void;
  copy?: (id: string) => void;
  checkbox?: boolean;
  handleCheckbox?: (templateItem: TemplateItem) => void;
  checkedItems?: string[];
}

const CheckboxContainer = styled.div`
  position: absolute;
  right: 0;
  top: 30%;
`;

const StyledButton = styled.button`
  margin-left: 154px;
`;

const ListCardItems: React.FunctionComponent<ListCardProps> = ({
  titleKey,
  listData,
  edit,
  remove,
  copy,
  checkbox,
  checkedItems,
  handleCheckbox,
}) => {
  const actionHandler = (assessmentId: string) => (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    if (event.currentTarget.name === 'edit') {
      if (edit) {
        edit(assessmentId);
      }
    } else if (event.currentTarget.name === 'delete') {
      if (remove) {
        remove(assessmentId);
      }
    } else if (event.currentTarget.name === 'copy') {
      if (copy) {
        copy(assessmentId);
      }
    }
  };

  const renderContent = (category: string, type: string, competency: string, itemStatus: string) => (
    <React.Fragment>
      <div className="col-md-3 p-l-0">
        <LabelGroup label="Category" value={category} />
      </div>
      <div className="col-md-3 border-left text-center">
        <LabelGroup label="Type" value={type} badge={BadgeTypes.PRIMARY} />
      </div>
      <div className="col-md-3 border-left text-center">
        <LabelGroup label="Status" value={itemStatus} badge={BadgeTypes.PRIMARY} />
      </div>
      {competency.length > 0 && (
        <div className="col-md-3 border-left text-center">
          <LabelGroup label="Competency" value={competency} />
        </div>
      )}
    </React.Fragment>
  );

  const renderActions = (id: string, item: any) => {
    const renderActionButton = (name: string, text: string, icon: IconProp, className: string) => (
      <IconButton name={name} icon={icon} className={className} actionHandler={actionHandler(id)}>
        {text}
      </IconButton>
    );

    const renderCopyActionButton = (name: string, text: string, icon: IconProp, className: string) => (
      <StyledButton name={name} className={classNames(['btn', className])} onClick={actionHandler(id)} id={id}>
        {icon && <FontAwesomeIcon icon={icon} />}
        &nbsp;
        {text}
      </StyledButton>
    );

    if (checkbox) {
      return (
        <CheckboxContainer className="pull-right">
          <Checkbox
            name="templateItem"
            value={id}
            isChecked={checkedItems && !!checkedItems.find(checkedItem => checkedItem === id)}
            onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
              e.preventDefault();
              if (handleCheckbox) {
                handleCheckbox(item);
              }
            }}
          >
            {' '}
          </Checkbox>
        </CheckboxContainer>
      );
    }

    if (USER_ROLE.isSuperAdmin() && item.isSystemDefined) {
      return (
        <React.Fragment>
          {edit && renderActionButton('edit', 'Edit', 'edit', 'btn-outline btn-primary')}
          {remove && renderActionButton('delete', 'Delete', 'trash', 'btn-default')}
          {edit && renderActionButton('copy', 'Copy', 'copy', 'btn-outline btn-primary')}
        </React.Fragment>
      );
    } else if (USER_ROLE.isClientAdmin() && item.isSystemDefined) {
      return (
        <React.Fragment>
          {edit && renderCopyActionButton('copy', 'Copy', 'copy', 'btn-outline btn-primary')}
        </React.Fragment>
      );
    } else {
      return (
        <React.Fragment>
          {edit && renderActionButton('edit', 'Edit', 'edit', 'btn-outline btn-primary')}
          {remove && renderActionButton('delete', 'Delete', 'trash', 'btn-default')}
          {edit && renderActionButton('copy', 'Copy', 'copy', 'btn-outline btn-primary')}
        </React.Fragment>
      );
    }
  };

  function renderAllCards(item: any) {
    const content =
      (item.category || item.type || item.competency) &&
      renderContent(item.category, item.type, item.competency, item.itemStatus);

    const actions = renderActions(item.id || item.itemId, item);

    return (
      <ItemCard key={item.id || item.itemId} header={item[titleKey]}>
        {{ content, actions }}
      </ItemCard>
    );
  }

  return <React.Fragment>{listData.length > 0 ? listData && listData.map(renderAllCards) : null}</React.Fragment>;
};

export default ListCardItems;
