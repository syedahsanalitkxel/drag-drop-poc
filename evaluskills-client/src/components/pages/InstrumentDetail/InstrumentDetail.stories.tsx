import React, { useState } from 'react';

import { storiesOf } from '@storybook/react';
import Instrument from '../../pages/InstrumentClient';
import { ClientInstruments } from '../../../interfaces/Instruments';
import InstrumentDetail from './index';
import AssessmentItemInterface from '../../../interfaces/AssessmentItem';
import ParticipantInterface from '../../../interfaces/Participant';
import EvaluatorInterface from '../../../interfaces/Evaluator';
import InstrumentDetailTemplate from '../../../containers/InstrumentDetailContainer';

const instruments = [
  {
    id: '1',
    noOfAssessmentItems: '25',
    noOfEvaluations: '28',
    noOfParticipants: '30',
    title: '360° Leadership Instrument',
  },
  {
    id: '2',
    noOfAssessmentItems: '25',
    noOfEvaluations: '28',
    noOfParticipants: '30',
    title: '210° Leadership Instrument',
  },
];

const AssessmentItems: AssessmentItemInterface[] = [
  {
    category: 'Character',
    competency: 'Team Player',
    definition: 'Receive feedback from others and uses the feedback to improve performance',
    id: 'uuid-12-321',
    type: 'Competency',
  },
  {
    category: 'Action',
    competency: 'Good Coder',
    definition: 'Has a set of moral principles used in job in accordance with the culture of organization',
    id: 'uuid-11-111',
    type: 'Influential',
  },
];

const participantsList: ParticipantInterface[] = [
  {
    email: 'jasminrasool@gmail.com',
    id: '1',
    name: 'jasmin Rasool',
    noOfEvaluators: '5',
    progress: '20%',
    status: 'InProgress',
  },
  {
    email: 'jasminrasool@gmail.com',
    id: '2',
    name: 'jasmin Rasool',
    noOfEvaluators: '2',
    progress: '50%',
    status: 'InProgress',
  },
  {
    email: 'jasminrasool@gmail.com',
    id: '3',
    name: 'jasmin Rasool',
    noOfEvaluators: '5',
    progress: '100%',
    status: 'Completed',
  },
  {
    email: 'jasminrasool@gmail.com',
    id: '4',
    name: 'jasmin Rasool',
    noOfEvaluators: '5',
    progress: '0%',
    status: 'NotStarted',
  },
  {
    email: 'jhonjames@gmail.com',
    id: '5',
    name: 'john james',
    noOfEvaluators: '5',
    progress: '20%',
    status: 'InProgress',
  },
  {
    email: 'jasminrasool@gmail.com',
    id: '6',
    name: 'jasmin Rasool',
    noOfEvaluators: '2',
    progress: '50%',
    status: 'InProgress',
  },
];

const EvaluatorList: EvaluatorInterface[] = [
  {
    email: 'robbyrash@gmail.com',
    id: '1',
    name: 'Robby Rash',
    progress: '20%',
    role: 'UI/UX Leader',
    status: 'InProgress',
  },
  {
    email: 'maraymrassol@gmail.com',
    id: '2',
    name: 'Mariyam Rassal',
    progress: '100%',
    role: 'Department Head',
    status: 'Completed',
  },
  {
    email: 'robbyrash@gmail.com',
    id: '3',
    name: 'Robby Rash',
    progress: '50%',
    role: 'Team Lead',
    status: 'InProgress',
  },
  {
    email: 'robbyrash@gmail.com',
    id: '4',
    name: 'Robby Rash',
    progress: '20%',
    role: 'Senior Designer',
    status: 'InProgress',
  },
];

function viewAssessmentDetail() {
  alert('Clicked detail button');
  // history.push('/client-assessment-detail');
}

storiesOf('InstrumentDetail', module).add('InstrumentDetail', () => (
  <InstrumentDetail instruments={instruments} participants={participantsList} AssessmentItems={AssessmentItems} />
));
