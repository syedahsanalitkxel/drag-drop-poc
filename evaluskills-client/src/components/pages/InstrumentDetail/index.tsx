import React, { useState } from 'react';

import InstrumentTemplateInterface from '../../../interfaces/InstrumentTemplate';
import PageBody from '../../atoms/PageBody';
import PageHeader from '../../atoms/PageHeader';
import Pager from '../../molecules/Pager';
import InstrumentDetailCard from '../../organisms/InstrumentDetailCard';
import DashboardTemplate from '../../templates/DashboardTemplate';
import { ClientInstruments } from '../../../interfaces/Instruments';
import AssessmentItemInterface from '../../../interfaces/AssessmentItem';
import ParticipantInterface from '../../../interfaces/Participant';
import EvaluatorInterface from '../../../interfaces/Evaluator';
import { Nav, NavItem, NavLink, TabContent, TabPane } from 'reactstrap';
import classnames from 'classnames';
import ListCardItems from '../../organisms/ListCardItems';

interface Props {
  instruments: any;
  view?: (evaluationId: string) => void;
  participants: ParticipantInterface[];
  evaluator: EvaluatorInterface[];
  assessmentItems: AssessmentItemInterface[];
}

const InstrumentDetailTemplate: React.FunctionComponent<Props> = ({
  instruments,
  view,
  participants,
  evaluator,
  assessmentItems,
}) => {
  console.log(instruments.title);
  const activateClickHandler = (event: React.MouseEvent) => {
    event.preventDefault();
    // toggleFilterModal();
  };

  const viewClickHandler = (event: React.MouseEvent) => {
    event.preventDefault();
    // toggleFilterModal();
  };

  const cancelClickHandler = (event: React.MouseEvent) => {
    event.preventDefault();
    // toggleFilterModal();
  };

  const [activeTab, setActiveTab] = useState('1');

  const toggle = () => {
    if (activeTab === '1') {
      setActiveTab('2');
    } else if (activeTab === '2') {
      setActiveTab('1');
    }
  };

  return (
    <DashboardTemplate>
      <div className="row">
        <div className="col-lg-12">
          <PageHeader
            title={instruments.title}
            activeButtonText="Activate"
            activeButtonActionHandler={activateClickHandler}
            viewButtonText="View Results"
            viewButtonActionHandler={viewClickHandler}
            cancelButtonText="Cancel Assessment"
            cancelButtonActionHandler={cancelClickHandler}
          />
          <PageBody>
            <Nav tabs={true}>
              <NavItem>
                <NavLink
                  id="1"
                  className={classnames({ active: activeTab === '1' })}
                  onClick={toggle}
                >
                  Participants {participants.length}
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  id="2"
                  className={classnames({ active: activeTab === '2' })}
                  onClick={toggle}
                >
                  Assessments Items {assessmentItems.length}
                </NavLink>
              </NavItem>
            </Nav>

            <TabContent activeTab={activeTab}>
              <TabPane tabId="1">
                <InstrumentDetailCard
                  titleKey="title"
                  participants={participants}
                  evaluator={evaluator}
                  view={view}
                />
              </TabPane>
              <TabPane tabId="2">
                <ListCardItems titleKey="definition" listData={assessmentItems} />
              </TabPane>
            </TabContent>
            {/*<Pager />*/}
          </PageBody>
        </div>
      </div>
    </DashboardTemplate>
  );
};

export default InstrumentDetailTemplate;
