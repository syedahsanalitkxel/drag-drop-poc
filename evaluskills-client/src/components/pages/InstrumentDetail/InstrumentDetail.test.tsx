import { configure, mount, render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import InstrumentDetail from './index';
import ParticipantInterface from '../../../interfaces/Participant';
import EvaluatorInterface from '../../../interfaces/Evaluator';
import AssessmentItemInterface from '../../../interfaces/AssessmentItem';
import { ClientInstruments } from '../../../interfaces/Instruments';
import InstrumentDetailCard from '../../organisms/InstrumentDetailCard';
import Participant from '../../../interfaces/Participant';
import { TabPane } from 'reactstrap';
import ListCardItems from '../../organisms/ListCardItems';

configure({ adapter: new Adapter() });
describe('Instrument Detail', () => {
  const AssessmentItems: AssessmentItemInterface[] = [
    {
      category: 'Character',
      competency: 'Team Player',
      definition: 'Receive feedback from others and uses the feedback to improve performance',
      id: 'uuid-12-321',
      type: 'Competency',
    },
    {
      category: 'Action',
      competency: 'Good Coder',
      definition: 'Has a set of moral principles used in job in accordance with the culture of organization',
      id: 'uuid-11-111',
      type: 'Influential',
    },
  ];

  const participantsList: ParticipantInterface[] = [
    {
      email: 'jasminrasool@gmail.com',
      id: '1',
      name: 'jasmin Rasool',
      noOfEvaluators: '5',
      progress: '20%',
      status: 'InProgress',
    },
    {
      email: 'jasminrasool@gmail.com',
      id: '2',
      name: 'jasmin Rasool',
      noOfEvaluators: '2',
      progress: '50%',
      status: 'InProgress',
    },
    {
      email: 'jasminrasool@gmail.com',
      id: '3',
      name: 'jasmin Rasool',
      noOfEvaluators: '5',
      progress: '100%',
      status: 'Completed',
    },
    {
      email: 'jasminrasool@gmail.com',
      id: '4',
      name: 'jasmin Rasool',
      noOfEvaluators: '5',
      progress: '0%',
      status: 'NotStarted',
    },
    {
      email: 'jhonjames@gmail.com',
      id: '5',
      name: 'john james',
      noOfEvaluators: '5',
      progress: '20%',
      status: 'InProgress',
    },
    {
      email: 'jasminrasool@gmail.com',
      id: '6',
      name: 'jasmin Rasool',
      noOfEvaluators: '2',
      progress: '50%',
      status: 'InProgress',
    },
  ];

  const EvaluatorList: EvaluatorInterface[] = [
    {
      email: 'robbyrash@gmail.com',
      id: '1',
      name: 'Robby Rash',
      progress: '20%',
      role: 'UI/UX Leader',
      status: 'InProgress',
    },
    {
      email: 'maraymrassol@gmail.com',
      id: '2',
      name: 'Mariyam Rassal',
      progress: '100%',
      role: 'Department Head',
      status: 'Completed',
    },
    {
      email: 'robbyrash@gmail.com',
      id: '3',
      name: 'Robby Rash',
      progress: '50%',
      role: 'Team Lead',
      status: 'InProgress',
    },
    {
      email: 'robbyrash@gmail.com',
      id: '4',
      name: 'Robby Rash',
      progress: '20%',
      role: 'Senior Designer',
      status: 'InProgress',
    },
  ];

  const InstrumentList: ClientInstruments[] = [
    {
      id: '1',
      noOfAssessmentItems: '25',
      noOfEvaluations: '28',
      noOfParticipants: '30',
      title: '360° Leadership Instrument',
    },
    {
      id: '2',
      noOfAssessmentItems: '25',
      noOfEvaluations: '28',
      noOfParticipants: '30',
      title: '210° Leadership Instrument',
    },
  ];

  const mockViewFunction = jest.fn();
  const filterMockFunction = jest.fn();

  const instrumentDetailProps = {
    assessmentItems: AssessmentItems,
    evaluator: EvaluatorList,
    instruments: InstrumentList,
    participants: participantsList,
    view: mockViewFunction,
  };

  const client = shallow(<InstrumentDetail {...instrumentDetailProps} />);

  it('render successfully', () => {
    expect(client).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <InstrumentDetail {...instrumentDetailProps} />
      </Router>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without crashing with props', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <InstrumentDetail {...instrumentDetailProps} />
      </Router>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Nav to be defined', () => {
    const wrapper = shallow(<InstrumentDetail {...instrumentDetailProps} />);
    expect(wrapper.find('Nav')).toBeDefined();
  });

  it('NavItem to be defined', () => {
    const wrapper = shallow(<InstrumentDetail {...instrumentDetailProps} />);
    expect(wrapper.find('NavItem')).toBeDefined();
  });

  it('NavLink to be defined', () => {
    const wrapper = shallow(<InstrumentDetail {...instrumentDetailProps} />);
    expect(wrapper.find('NavLink')).toBeDefined();
  });

  it('TabContent to be defined', () => {
    const wrapper = shallow(<InstrumentDetail {...instrumentDetailProps} />);
    expect(wrapper.find('TabContent')).toBeDefined();
  });

  it('TabPane to be defined', () => {
    const wrapper = shallow(<InstrumentDetail {...instrumentDetailProps} />);
    expect(wrapper.find('TabPane')).toBeDefined();
  });

  describe('InstrumentListCard', () => {
    const Key = 'title';

    const viewMockFunction = jest.fn();

    const props = {
      titleKey: Key,
      participants: participantsList,
      evaluator: EvaluatorList,
      view: viewMockFunction,
    };

    it('Instrument List Card to be defined', () => {
      const wrapper = shallow(<InstrumentDetailCard {...instrumentDetailProps} />);
      expect(wrapper.find('InstrumentDetailCard')).toBeDefined();
    });
    it('Instrument Detail Card Render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <Router>
          <InstrumentDetailCard />
        </Router>,
        div
      );
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Instrument Detail Card Render with props', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <Router>
          <InstrumentDetailCard {...props} />
        </Router>,
        div
      );
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Progress bar to be defined', () => {
      const wrapper = shallow(<InstrumentDetailCard {...props} />);
      expect(wrapper.find('.progress')).toBeDefined();
    });
  });

  describe('ListCardItems', () => {
    const Key = 'title';

    const props = { titleKey: Key, listData: AssessmentItems };

    it('List Card Item to be defined', () => {
      const wrapper = shallow(<ListCardItems {...instrumentDetailProps} />);
      expect(wrapper.find('ListCardItems')).toBeDefined();
    });
    it('List Card Item Render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <Router>
          <ListCardItems />
        </Router>,
        div
      );
      ReactDOM.unmountComponentAtNode(div);
    });

    it('List Card Item Render with props', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <Router>
          <ListCardItems {...props} />
        </Router>,
        div
      );
      ReactDOM.unmountComponentAtNode(div);
    });
  });
});
