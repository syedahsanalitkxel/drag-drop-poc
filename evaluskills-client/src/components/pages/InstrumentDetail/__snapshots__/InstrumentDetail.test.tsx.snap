// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instrument Detail render successfully 1`] = `
<DashboardTemplate>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <PageHeader
        activeButtonActionHandler={[Function]}
        activeButtonText="Activate"
        cancelButtonActionHandler={[Function]}
        cancelButtonText="Cancel Assessment"
        viewButtonActionHandler={[Function]}
        viewButtonText="View Results"
      />
      <PageBody
        card={false}
      >
        <Nav
          tabs={true}
          tag="ul"
          vertical={false}
        >
          <NavItem
            tag="li"
          >
            <NavLink
              className="active"
              id="1"
              onClick={[Function]}
              tag="a"
            >
              Participants 
              6
            </NavLink>
          </NavItem>
          <NavItem
            tag="li"
          >
            <NavLink
              className=""
              id="2"
              onClick={[Function]}
              tag="a"
            >
              Assessments Items 
              2
            </NavLink>
          </NavItem>
        </Nav>
        <TabContent
          activeTab="1"
          tag="div"
        >
          <TabPane
            tabId="1"
            tag="div"
          >
            <InstrumentListCard
              evaluator={
                Array [
                  Object {
                    "email": "robbyrash@gmail.com",
                    "id": "1",
                    "name": "Robby Rash",
                    "progress": "20%",
                    "role": "UI/UX Leader",
                    "status": "InProgress",
                  },
                  Object {
                    "email": "maraymrassol@gmail.com",
                    "id": "2",
                    "name": "Mariyam Rassal",
                    "progress": "100%",
                    "role": "Department Head",
                    "status": "Completed",
                  },
                  Object {
                    "email": "robbyrash@gmail.com",
                    "id": "3",
                    "name": "Robby Rash",
                    "progress": "50%",
                    "role": "Team Lead",
                    "status": "InProgress",
                  },
                  Object {
                    "email": "robbyrash@gmail.com",
                    "id": "4",
                    "name": "Robby Rash",
                    "progress": "20%",
                    "role": "Senior Designer",
                    "status": "InProgress",
                  },
                ]
              }
              participants={
                Array [
                  Object {
                    "email": "jasminrasool@gmail.com",
                    "id": "1",
                    "name": "jasmin Rasool",
                    "noOfEvaluators": "5",
                    "progress": "20%",
                    "status": "InProgress",
                  },
                  Object {
                    "email": "jasminrasool@gmail.com",
                    "id": "2",
                    "name": "jasmin Rasool",
                    "noOfEvaluators": "2",
                    "progress": "50%",
                    "status": "InProgress",
                  },
                  Object {
                    "email": "jasminrasool@gmail.com",
                    "id": "3",
                    "name": "jasmin Rasool",
                    "noOfEvaluators": "5",
                    "progress": "100%",
                    "status": "Completed",
                  },
                  Object {
                    "email": "jasminrasool@gmail.com",
                    "id": "4",
                    "name": "jasmin Rasool",
                    "noOfEvaluators": "5",
                    "progress": "0%",
                    "status": "NotStarted",
                  },
                  Object {
                    "email": "jhonjames@gmail.com",
                    "id": "5",
                    "name": "john james",
                    "noOfEvaluators": "5",
                    "progress": "20%",
                    "status": "InProgress",
                  },
                  Object {
                    "email": "jasminrasool@gmail.com",
                    "id": "6",
                    "name": "jasmin Rasool",
                    "noOfEvaluators": "2",
                    "progress": "50%",
                    "status": "InProgress",
                  },
                ]
              }
              titleKey="title"
              view={[MockFunction]}
            />
          </TabPane>
          <TabPane
            tabId="2"
            tag="div"
          >
            <ListCardItems
              listData={
                Array [
                  Object {
                    "category": "Character",
                    "competency": "Team Player",
                    "definition": "Receive feedback from others and uses the feedback to improve performance",
                    "id": "uuid-12-321",
                    "type": "Competency",
                  },
                  Object {
                    "category": "Action",
                    "competency": "Good Coder",
                    "definition": "Has a set of moral principles used in job in accordance with the culture of organization",
                    "id": "uuid-11-111",
                    "type": "Influential",
                  },
                ]
              }
              titleKey="definition"
            />
          </TabPane>
        </TabContent>
        <Pager />
      </PageBody>
    </div>
  </div>
</DashboardTemplate>
`;
